packages/@aws-cdk/aws-kms/lib/alias.ts

```diff
@@ -137,6 +137,22 @@ export class Alias extends AliasBase {
     return new _Alias(scope, id);
   }
 
+  /**
+   * Import an existing KMS Alias defined outside the CDK app, by the alias name. This Alias will not
+   * have a direct reference to the KMS Key, so addAlias and grant* methods are not supported.
+   *
+   * @param scope The parent creating construct (usually `this`).
+   * @param id The construct's name.
+   * @param aliasName The full name of the KMS Alias (e.g., 'alias/aws/s3', 'alias/myKeyAlias').
+   */
+  public static fromAliasName(scope: Construct, id: string, aliasName: string): IAlias {
+    class Import extends AliasBase {
+      public get aliasName() { return aliasName; }
+      public get aliasTargetKey(): IKey { throw new Error('Cannot access aliasTargetKey on an Alias imnported by Alias.fromAliasName().'); }
```

Comment:
I think we need to return an actual implementation of `IKey` here, which does the following:

1. `keyId` returns `aliasName`.
2. `keyArn` also returns `aliasName` (this one is controversial, but I think it's the correct thing to do).
3. `addAlias()` throws.
4. `addToResourcePolicy()` returns an unsuccessful add.
5. `grant*()` return un unsuccessful add.

Reply:
I think for `addToResourcePolicy` all you have to do is return something like `{ statementAdded: false }`.

Not sure that I agree with the grant... grants generally throw if they can't accomplish what you asked them to do, and in this case it's strictly impossible to grant these permissions.

Reply:
Playing devil's advocate: if `addAlias` and `grant*` are going to throw, and addToResourcePolicy act as a no-op, what's the advantage of returning an `IKey` over the current implementation? There are no active uses of `aliasTargetKey` within the CDK itself (outside of this file), so there is no impact to other Constructs. For customer code, this is new functionality that is going to need to be explicitly adopted. What's the use case for needing to import a key by alias, and then wanting to access `myAlias.aliasTargetKey.keyId` rather than `myAlias.keyId`? I have a (slight) preference for the "fail fast" approach here of being very explicit that there is no underlying key, rather than having some unexpected functionality (e.g. `myAlias.aliasTargetKey.keyArn === myAlias.aliasName`).

Reply:
I don't see much advantage in returning correctly from `myAlias.keyId`, but suddenly have `myAlias.aliasTargetKey.keyId` throw. You also don't control what other constructs do in their implementation, so you're limiting the usefulness of this method considerably (if a customer gets this exception, there's basically nothing they can do to unblock themselves).

`myAlias.aliasTargetKey.keyArn === myAlias.aliasName` is already a problem, as we have to decide what to do with the `myAlias.keyArn` property.

For `grant*` if we have a method in `Grant` that does exactly what we need here: [`Grant.drop`](https://docs.aws.amazon.com/cdk/api/latest/docs/@aws-cdk_aws-iam.Grant.html#static-dropgrantee-intent).

In my opinion, if we throw from grants here, this method will be close to useless (you won't be able to pass this `IKey` anywhere as a construct argument, because the simplest `grantRead()` on an S3 Bucket / `grantEncrypt()` will immediately fail).

Reply:
I guess making `aliasTargetKey` throw now and then changing it later will be a backwards-compatible change, so I'm fine with leaving it as throwing for now and returning to this topic if we get some customer feedback that this is irritating.

I still think `grant*()` methods shouldn't throw though.

Reply:
Makes sense, I'll take this approach of addAlias/aliasTargetKey throwing, and the rest no-op-ing.